tests/test_model_manager.py

```diff
@@ -0,0 +1,294 @@
+# Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from .common import test_result_collector as trc
+
+from .mocks.mock_config import MockConfig
+from .mocks.mock_os import MockOSMethods
+from .mocks.mock_model_config import MockModelConfig
+from .mocks.mock_client import MockTritonClientMethods
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile
+from model_analyzer.config.run.run_search import RunSearch
+from model_analyzer.config.run.run_config_generator import RunConfigGenerator
+from model_analyzer.constants import LOGGER_NAME
+from model_analyzer.model_manager import ModelManager
+from model_analyzer.state.analyzer_state_manager import AnalyzerStateManager
+
+import itertools
+import logging
+import sys
+
+from unittest.mock import MagicMock
+from unittest.mock import patch
+
+
+
+class ModelManagerSubclass(ModelManager):
+    """ 
+    Overrides execute_run_configs() to gather a list of tuples that contain 
+    the main values of each 'executed' run_config
+    """
+
+    def __init__(self, config, client, server, metrics_manager, result_manager, state_manager):
+        super().__init__(config, client, server, metrics_manager, result_manager, state_manager)
+        self._configs = []
+    
+    def _execute_run_configs(self):
+        while self._run_config_generator.run_configs():
+            config = self._run_config_generator.next_config()
+            model_config = config.model_config().to_dict()
+            perf_config = config.perf_config()
+
+            instances = None
+            if model_config.get("instanceGroup") is not None:
+                instances = model_config["instanceGroup"][0]["count"]
+
+            dynamic_batching = None
+            if model_config.get("dynamicBatching") is not None:
+                dynamic_batching = model_config["dynamicBatching"].get("preferredBatchSize", [0])[0]
+
+            batch_size = perf_config.__getitem__("batch-size")
+            concurrency = perf_config.__getitem__("concurrency-range")
+
+            self._configs.append((instances, dynamic_batching, batch_size, concurrency))
+
+    def get_run_configs(self):
+        return self._configs
+
+
```

Comment:
can't comment on the line below because Elias already put a thread on there and GitHub is weird

no need to specify `MagicMock()` as the next argument--it's the default:

https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch

Reply:
I'll keep that in mind next time. While we are still learning this all I think it is best to be explicit

Reply:
Also, this doesn't work as-is.  If I omit the `MagicMock()` I would then need to pass in an extra arg into every test function (I confirmed that all the tests break without further edits):

_If patch() is used as a decorator and new is omitted, the created mock is passed in as an extra argument to the decorated function._